### PR TITLE
Improve Python setup on Travis

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Had to restart one job, passed now, but flaky stage was not run.
I.e. this should be considered to be looking green.. :)

**Review by jamessan — APPROVED** [review]

**Comment by blueyed** [comment]
@justinmk 
Do you like it also?

File: ci/before_install.sh
Comment (justinmk) [inline ci/before_install.sh]: if `python3 --version` fails does it continue to the next line?

File: ci/before_install.sh
Comment (justinmk) [inline ci/before_install.sh]: using pyenv was already tried before: https://github.com/neovim/neovim/pulls?q=is%3Apr+pyenv+is%3Aclosed

It doesn't really fix the problems with python on travis. Are you sure it works now?

File: ci/install.sh
Comment (justinmk) [inline ci/install.sh]: I do not understand this change. The `-m pip` form was very intentional...

File: ci/before_install.sh
Comment (blueyed) [reply ci/before_install.sh]: Yes.. see the build logs.
I've checked osx and linux before merging.

File: ci/before_install.sh
Comment (blueyed) [reply ci/before_install.sh]: > I really need to look at these PRs before they are merged.

I see, but I've waited several days and figured it would be good to move on.

(this was related to progressing on Python/pynvim setup in general)

I will be more passive here in the future I guess.

Comment (blueyed) [reply ci/before_install.sh]: As for the pyenv setup itself: it is more straight forward now in my opinion.

File: ci/install.sh
Comment (blueyed) [reply ci/install.sh]: It is just more straight forward IMHO - it really ensures that pip is also installed in the path.

File: ci/before_install.sh
Comment (justinmk) [reply ci/before_install.sh]: More straightfoward how? `python -m pip` is the python-recommended way of invoking `pip`. Using `pip` directly is deprecated, and it is imprecise because different versions of python may exist on a system.

File: ci/install.sh
Comment (justinmk) [reply ci/install.sh]: `python -m pip` is the python-recommended way of invoking `pip[3]`. Using `pip[3]` directly is deprecated, and  imprecise because different versions of python may exist on a system.

When reverting changes that were made previously (see `git log`) it is good to at least discuss it explicitly with the previous author.

File: ci/before_install.sh
Comment (justinmk) [reply ci/before_install.sh]: > I will be more passive here in the future I guess.

It's good to be active but when reverting previous change we usually try to get signoff by the previous author. Otherwise it causes circular changes.

File: ci/install.sh
Comment (blueyed) [reply ci/install.sh]: Ok, will change that.

File: ci/before_install.sh
Comment (blueyed) [reply ci/before_install.sh]: > More straightfoward how? 

It explicitly uses it, and hsa python3/pip3 now installed always.

Comment (justinmk) [reply ci/before_install.sh]: Check `git log --grep pyenv` , we tried this in the past and it didn't work, because of an unresolved Travis issue.

**Pull request merged**
